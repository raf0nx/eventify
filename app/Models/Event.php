<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Event extends Model {
    use HasFactory;

    protected $fillable = [
        'name',
        'description',
        'image',
        'start_datetime',
    ];

    public function users(): \Illuminate\Database\Eloquent\Relations\BelongsToMany{
        return $this->belongsToMany(User::class, 'users_events');
    }
}
