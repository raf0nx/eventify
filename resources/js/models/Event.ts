import { User } from "./User";

export interface Event {
    created_at: Date;
    description: string;
    event_image: string;
    id: number;
    name: string;
    start_datetime: Date;
    updated_at: Date;
    users: User[];
}
