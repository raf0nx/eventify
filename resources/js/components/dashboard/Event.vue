<template>
	<v-lazy
		v-model="isActive"
		:options="{
			threshold: 0.5,
		}"
        min-height="200"
		transition="fade-transition"
	>
		<v-card class="my-12" elevation="8">
			<v-img
				lazy-src="https://picsum.photos/id/11/10/6"
				src="https://picsum.photos/id/11/500/300"
				height="300"
			></v-img>

			<v-card-subtitle class="pb-0">
				Event starts on {{ eventStartDate }} at {{ eventStartTime }}
			</v-card-subtitle>

			<v-card-title class="font-weight-bold">{{
				event.name
			}}</v-card-title>

			<v-card-text>
				<div>
					{{ event.description }}
				</div>

				<div class="pt-4">
					{{ event.users.length }} interested &bull; 0 going
				</div>
			</v-card-text>

			<v-card-actions class="py-4">
				<v-btn color="deep-purple" text>
					<v-icon left>mdi-star</v-icon>
					Interested
				</v-btn>
				<v-btn color="deep-purple" class="white--text font-weight-bold">
					<v-icon left>mdi-check-circle</v-icon>
					Going
				</v-btn>

				<v-spacer></v-spacer>

				<v-btn icon>
					<v-tooltip top>
						<template v-slot:activator="{ on, attrs }">
							<v-icon v-on="on" v-bind="attrs">mdi-heart</v-icon>
						</template>
						<span>Add to favourites</span>
					</v-tooltip>
				</v-btn>

				<v-btn icon>
					<v-tooltip top>
						<template v-slot:activator="{ on, attrs }">
							<v-icon v-on="on" v-bind="attrs">
								mdi-share-variant
							</v-icon>
						</template>
						<span>Share</span>
					</v-tooltip>
				</v-btn>
			</v-card-actions>
		</v-card>
	</v-lazy>
</template>

<script lang="ts">
	import { Vue, Component, Prop } from "vue-property-decorator";
	import moment from "moment";

	import { Event as EventModel } from "@/models/Event";

	@Component
	export default class Event extends Vue {
		@Prop()
		event!: EventModel;

        isActive = false;

		get eventStartDate(): string {
			return moment(this.event.start_datetime).format("dddd, MMM Do YYYY");
		}

		get eventStartTime(): string {
			return moment(this.event.start_datetime).format("h a");
		}
	}
</script>

<style>
</style>