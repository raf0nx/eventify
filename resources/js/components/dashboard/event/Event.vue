<template>
	<v-col class="col-xs-12 col-sm-6 col-lg-4 col-xl-3">
		<v-lazy
			v-model="isActive"
			:options="{
				threshold: 0.5,
			}"
			min-height="240"
			transition="fade-transition"
		>
			<v-hover v-slot="{ hover }">
				<v-card
					ripple
					class="rounded-lg"
					:elevation="hover ? 12 : 4"
					max-height="480"
					@click="goToEventDetails()"
				>
					<v-img
						lazy-src="https://picsum.photos/id/11/10/6"
						src="https://picsum.photos/id/11/500/300"
						max-height="200"
						class="rounded-t-lg"
					></v-img>

					<v-card-subtitle class="pb-0">
						{{ eventStartDate }} at {{ eventStartTime }}
					</v-card-subtitle>

					<v-card-title class="font-weight-bold">
						<router-link
							:to="{ name: 'Event', params: { id: event.id } }"
							class="
								event__link
								black--text
								text-decoration-none text-truncate
							"
							>{{ event.name }}</router-link
						>
					</v-card-title>

					<v-card-text>
						<v-tooltip bottom max-width="320">
							<template v-slot:activator="{ on, attrs }">
								<div
									class="text-truncate"
									v-on="on"
									v-bind="attrs"
								>
									{{ event.description }}
								</div>
							</template>
							<p class="text-justify mb-0">
								{{ event.description }}
							</p>
						</v-tooltip>
						<div class="pt-4">
							{{ event.users.length }} interested &bull; 0 going
						</div>
					</v-card-text>

					<v-card-actions class="pt-2 pb-4">
						<v-btn
							@click.stop="() => true"
							color="deep-purple"
							text
						>
							<v-icon left>mdi-star</v-icon>
							Interested
						</v-btn>
						<v-btn
							@click.stop="() => true"
							color="deep-purple"
							class="white--text font-weight-bold"
						>
							<v-icon left>mdi-check-circle</v-icon>
							Going
						</v-btn>

						<v-spacer></v-spacer>

						<v-btn icon>
							<v-tooltip top>
								<template v-slot:activator="{ on, attrs }">
									<v-icon
										@click.stop="() => true"
										color="red accent-2"
										v-on="on"
										v-bind="attrs"
										>mdi-heart-outline</v-icon
									>
								</template>
								<span>Add to favourites</span>
							</v-tooltip>
						</v-btn>
					</v-card-actions>
				</v-card>
			</v-hover>
		</v-lazy>
	</v-col>
</template>

<script lang="ts">
	import { Vue, Component, Prop } from "vue-property-decorator";
	import moment from "moment";

	import { Event as EventModel } from "@/models/Event";

	@Component
	export default class Event extends Vue {
		@Prop()
		event!: EventModel;

		isActive = false;

		get eventStartDate(): string {
			return moment(this.event.start_datetime).format("dddd, MMM Do YYYY");
		}

		get eventStartTime(): string {
			return moment(this.event.start_datetime).format("h a");
		}

		goToEventDetails(): void {
			setTimeout(() => {
				this.$router.push({
					name: "Event",
					params: { id: String(this.event.id) },
				});
			}, 200);
		}
	}
</script>

<style scoped>
	.event__link {
		border-bottom: 2px solid white;
		transition: border-bottom linear 0.5s;
	}
	.event__link:hover {
		border-bottom: 2px solid black;
	}
</style>