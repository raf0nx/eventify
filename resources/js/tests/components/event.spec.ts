import { createLocalVue, shallowMount, Wrapper } from "@vue/test-utils";
import Vuetify from "vuetify";

import Event from "@/components/dashboard/event/Event.vue";
import { Event as EventMock } from "@/tests/constans/Event";

describe("Dashboard.vue", () => {
    const localVue = createLocalVue();
    let wrapper: Wrapper<Event>;
    let vuetify: Vuetify;

    beforeEach(() => {
        vuetify = new Vuetify();
        wrapper = shallowMount(Event, {
            localVue,
            vuetify,
            propsData: { EventMock }
        });
    });

    afterEach(() => {
        wrapper.destroy();
    });

    it("Should match snapshot", () => {
        // Assert
        expect(wrapper.html()).toMatchSnapshot();
    });
});
