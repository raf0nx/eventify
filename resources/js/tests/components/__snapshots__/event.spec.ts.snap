// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event.vue Should match snapshot 1`] = `
"<v-lazy-stub minheight=\\"200\\" options=\\"[object Object]\\" tag=\\"div\\" transition=\\"fade-transition\\">
  <v-card-stub loaderheight=\\"4\\" tag=\\"div\\" elevation=\\"8\\" class=\\"my-12\\">
    <v-img-stub height=\\"320\\" lazysrc=\\"https://picsum.photos/id/11/10/6\\" options=\\"[object Object]\\" position=\\"center center\\" src=\\"https://picsum.photos/id/11/500/300\\" transition=\\"fade-transition\\"></v-img-stub>
    <v-card-subtitle-stub class=\\"pb-0\\">
      Event starts on Friday, Jan 1st 2021 at 12 am
    </v-card-subtitle-stub>
    <v-card-title-stub class=\\"font-weight-bold\\">event</v-card-title-stub>
    <v-card-text-stub>
      <div>
        Lorem ipsum dolor sit amet, consectetur adip
      </div>
      <div class=\\"pt-4\\">
        1 interested â€¢ 0 going
      </div>
    </v-card-text-stub>
    <v-card-actions-stub class=\\"py-4\\">
      <v-btn-stub color=\\"deep-purple\\" tag=\\"button\\" activeclass=\\"\\" text=\\"true\\" type=\\"button\\">
        <v-icon-stub left=\\"\\">mdi-star</v-icon-stub>
        Interested
      </v-btn-stub>
      <v-btn-stub color=\\"deep-purple\\" tag=\\"button\\" activeclass=\\"\\" type=\\"button\\" class=\\"white--text font-weight-bold\\">
        <v-icon-stub left=\\"\\">mdi-check-circle</v-icon-stub>
        Going
      </v-btn-stub>
      <v-spacer-stub></v-spacer-stub>
      <v-btn-stub tag=\\"button\\" activeclass=\\"\\" icon=\\"true\\" type=\\"button\\">
        <v-tooltip-stub opendelay=\\"0\\" closedelay=\\"0\\" contentclass=\\"\\" fixed=\\"true\\" top=\\"true\\" openonhover=\\"true\\" maxwidth=\\"auto\\" nudgebottom=\\"0\\" nudgeleft=\\"0\\" nudgeright=\\"0\\" nudgetop=\\"0\\" nudgewidth=\\"0\\" tag=\\"span\\"> <span>Add to favourites</span></v-tooltip-stub>
      </v-btn-stub>
      <v-btn-stub tag=\\"button\\" activeclass=\\"\\" icon=\\"true\\" type=\\"button\\">
        <v-tooltip-stub opendelay=\\"0\\" closedelay=\\"0\\" contentclass=\\"\\" fixed=\\"true\\" top=\\"true\\" openonhover=\\"true\\" maxwidth=\\"auto\\" nudgebottom=\\"0\\" nudgeleft=\\"0\\" nudgeright=\\"0\\" nudgetop=\\"0\\" nudgewidth=\\"0\\" tag=\\"span\\"> <span>Share</span></v-tooltip-stub>
      </v-btn-stub>
    </v-card-actions-stub>
  </v-card-stub>
</v-lazy-stub>"
`;
